Core/src/org/sleuthkit/autopsy/modules/hashdatabase/HashDbManager.java

```diff
@@ -582,16 +693,91 @@ public String getDisplayName() {
 
             INDEXING_DONE
         }
+        
+        public abstract String getHashSetName();
+        
+        abstract String getDisplayName();
+
+        public abstract String getDatabasePath() throws TskCoreException;
+
+        public abstract SleuthkitHashSet.KnownFilesType getKnownFilesType();
+
+        public abstract boolean getSearchDuringIngest();
+        
+        abstract boolean getDefaultSearchDuringIngest();
```

Comment:
I'm thinking the default is a property of the type of hash set, and the flag should be initialized for each concrete hash set type in the constructors of the corresponding class, eliminating the need for this method. Am I missing something?

Reply:
I added this in preparation for checking whether the database is locally created. In a coming PR, getDefaultSearchDuringIngest for central repo queries the properties file for the list of names of locally created databases and returns true or false based on whether the database is found. That settings file I believe will not be updated when the HashDb object is created (it waits until Ok/Apply).